### PR TITLE
docs glossary

### PR DESCRIPTION
**Pull request by piotrm0**
Items to add to release announcement:
- **Glossary**: added a glossary of terms in the documentation on the index page of the Core Concepts section.

### CODE REVIEW

File: docs/trulens_eval/getting_started/core_concepts/index.md [hunk 1]
```diff
@@ -1,5 +1,90 @@
...
+
+- `Feedback Function`. A method that implements an `Evaluation`. This
+  corresponds to [Feedback][trulens_eval.feedback.feedback.Feedback].
+
+- `Generation`. See `Completion`.
+
+- `Human Feedback`. A feedback that is provided by a human, e.g. a thumbs
+  up/down in response to a `Completion`.
+
+- `LLM`, `Large Language Model`. The `Component` of an `Application` that
```
Comment (joshreini1) [inline docs/trulens_eval/getting_started/core_concepts/index.md:51]: Skip this definition? LLMs do a lot (even in our sense) more than completion

File: Makefile [hunk 2]
```diff
@@ -18,9 +18,14 @@ lab:
 	$(CONDA) .conda/docs; jupyter lab --ip=0.0.0.0 --no-browser --ServerApp.token=deadbeef
 
 # Serve the documentation website.
```
Comment (joshreini1) [inline Makefile:20]: how is the make file used? do we have docs for this?

**Review by joshreini1 — CHANGES_REQUESTED** [review]

File: docs/trulens_eval/getting_started/core_concepts/index.md (hunk 1 above)
Comment (piotrm0) [reply docs/trulens_eval/getting_started/core_concepts/index.md:51]: Do you mean in non-strictly-technical definition of the term? The technical definition comes from "language model" which is exactly a next token prediction model, i.e. a model that completes a piece of text.

File: Makefile (hunk 2 above)
Comment (piotrm0) [reply Makefile:23]: No but I added some text to each Makefile in this PR just now.

File: docs/trulens_eval/getting_started/core_concepts/index.md (hunk 1 above)
Comment (piotrm0) [reply docs/trulens_eval/getting_started/core_concepts/index.md:51]: (What is the non-strictly-technical part of the definition that it is missing?)

Comment (joshreini1) [reply docs/trulens_eval/getting_started/core_concepts/index.md:51]: Yeah, in the non-strictly-technical sense, it's used for *eval*, tool selection, routing, etc. Get your point though, that these are still completion tasks

**Review by joshreini1 — APPROVED** [review]

**Pull request merged**